go-controller/pkg/node/gateway_init.go

```diff
@@ -513,6 +527,37 @@ func (nc *DefaultNodeNetworkController) initGatewayDPUHost(kubeNodeIP net.IP) er
 		return err
 	}
 
+	nodeIPNetv4, _ := util.MatchFirstIPNetFamily(false, ifAddrs)
+	nodeAddrSet := sets.New[string](nodeIPNetv4.String())
+
+	if gatewayIntfTemp != "" && gatewayIntfTemp != gatewayIntf {
+		config.Gateway.Interface = gatewayIntfTemp
```

Comment:
didn't you just assign
`gatewayIntfTemp := config.Gateway.Interface`? and now the other way around?

Reply:
as discussed offline we overwrite it here:
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5307/files#diff-e9790f3508dd0c36e1fd2634870b3066acebe3d2d370d7c9e276ddb8e2278b21R519

and we are dancing around it because of the usage of: `getGatewayNextHops`